src/include/catalog/default_value.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <string>
+#include <utility>
+
+#include "type/type_id.h"
+
+namespace terrier::catalog {
+/**
+ * Wraps the concept of a default value expression into an object that can be
+ * passed between the catalog and the consumers of these expressions.
+ */
+class DefaultValue {
```

Comment:
Is this not the same type of class as `CatalogAccessor::ColumnDefinition` (i.e., it's information that can be decoupled from the `Schema`)?

If yes, then we should rename this class to `DefaultValueDefinition` and maybe store it **src/include/catalog/definitions/** with `ColumnDefinition` and `IndexKeyDefinition`.

Reply:
I’ll admit that I need some mental help on what the right internal representation for default values are, but this is different from the other definition classes because the default values are independent of the catalog (whereas the other’s get defined by user code and assigned OIDs when submitted).

Maybe this should live under a different component?